utilities/checkpoint/checkpoint_test.cc

```diff
@@ -289,6 +296,95 @@ TEST_F(CheckpointTest, GetSnapshotLink) {
   }
 }
 
+TEST_F(CheckpointTest, ExportColumnFamilyWithLinks) {
+    const auto export_path = test::TmpDir(env_) + "/export";
+    auto options = CurrentOptions();
+    delete db_;
+    db_ = nullptr;
+    ASSERT_OK(DestroyDB(dbname_, options));
+    ASSERT_OK(DestroyDB(export_path, options));
+    test::DestroyDir(env_, export_path);
+
+    // Create a database
+    Status s;
+    options.create_if_missing = true;
+    ASSERT_OK(DB::Open(options, dbname_, &db_));
+    const auto key = std::string("foo");
+    ASSERT_OK(Put(key, "v1"));
+
+    Checkpoint* checkpoint;
+    ASSERT_OK(Checkpoint::Create(db_, &checkpoint));
+
+    // Export the Tables
+    std::vector<LiveFileMetaData> metadata;
+    // Verify there is one sst file in metadata and export directory.
+    auto verify_files_exported = [&](int num_files_expected) {
+      ASSERT_EQ(metadata.size(), num_files_expected);
+      std::vector<std::string> subchildren;
+      env_->GetChildren(export_path, &subchildren);
+      int num_children = 0;
+      for (const auto& child : subchildren) {
+        if (child != "." && child != "..") {
+          ++num_children;
+        }
+      }
+      ASSERT_EQ(num_children, num_files_expected);
+    };
+
+    ASSERT_OK(checkpoint->ExportColumnFamily(db_->DefaultColumnFamily(),
+                                             &metadata, export_path));
+    verify_files_exported(1);
+    test::DestroyDir(env_, export_path);
+
+    // Check after compaction
+    CompactAll();
+    ASSERT_OK(Put(key, "v2"));
+    metadata.clear();
+    ASSERT_OK(checkpoint->ExportColumnFamily(db_->DefaultColumnFamily(),
+                                             &metadata, export_path));
+    verify_files_exported(2);
+    test::DestroyDir(env_, export_path);
```

Comment:
I know we don't always follow this but a better place to do the cleaning up is in the destructor of the test class. The code here will not be executed if we hit an assert failure above. If it's not too hard for you to move to the destructor, it's worth doing.

Reply:
Done.